src/main/java/org/jetbrains/bsp/bazel/server/bsp/services/ScalaBuildServerService.java

```diff
@@ -39,19 +55,64 @@ public ScalaBuildServerService(BazelData bazelData, BazelRunner bazelRunner) {
 
   public Either<ResponseError, ScalacOptionsResult> buildTargetScalacOptions(
       ScalacOptionsParams scalacOptionsParams) {
-
     List<ScalacOptionsItem> resultItems =
         targetsLanguageOptionsResolver.getResultItemsForTargets(scalacOptionsParams.getTargets());
 
     ScalacOptionsResult javacOptionsResult = new ScalacOptionsResult(resultItems);
     return Either.forRight(javacOptionsResult);
   }
 
-  public CompletableFuture<ScalaTestClassesResult> buildTargetScalaTestClasses(
+  public Either<ResponseError, ScalaTestClassesResult> buildTargetScalaTestClasses(
       ScalaTestClassesParams scalaTestClassesParams) {
-    System.out.printf("DWH: Got buildTargetScalaTestClasses: %s%n", scalaTestClassesParams);
-    // TODO(illicitonion): Populate
-    return CompletableFuture.completedFuture(new ScalaTestClassesResult(new ArrayList<>()));
+    TargetRulesResolver<ScalaTestClassesItem> targetRulesResolver =
+        TargetRulesResolver.withBazelRunnerAndFilterAndMapper(
+            bazelRunner, this::isScalaTestRule, this::mapRuleToTestClassesItem);
+
+    List<ScalaTestClassesItem> resultItems =
+        targetRulesResolver.getItemsForTargets(scalaTestClassesParams.getTargets());
+
+    ScalaTestClassesResult scalaTestClassesResult = new ScalaTestClassesResult(resultItems);
+
+    return Either.forRight(scalaTestClassesResult);
+  }
+
+  private ScalaTestClassesItem mapRuleToTestClassesItem(Build.Rule rule) {
+    BuildTargetIdentifier target = new BuildTargetIdentifier(rule.getName());
+    List<String> classes = getTestClasses(rule);
+
+    return new ScalaTestClassesItem(target, classes);
+  }
+
+  private List<String> getTestClasses(Build.Rule rule) {
+    List<String> mainClasses = getTestMainClasses(rule);
+    List<String> srcsClasses = getTestSrcsClasses(rule);
+
+    return Stream.concat(mainClasses.stream(), srcsClasses.stream()).collect(Collectors.toList());
```

Comment:
`main_class` is the defined name to the class where the main is, while `srcs` has the path to all the source files in a given rule. I'm not sure what you're trying to do here, but it doesn't seem correct. Can you pls explain?

Reply:
More info on those values here: https://docs.bazel.build/versions/master/be/java.html#java_test

This is the docs for java, but all languages must abide by these rules

Reply:
I checked on more time and I agree that I misunderstood how `srcs` attribute works. But, now I found out that I should handle `test_class` attribute as well.

Reply:
I wanted to check what would happen if both attributes were present and how to approach this and ended up finding out that `rules_scala` doesn't actually support `test_class`.

Reply:
oh great, soo should I remove `srcsClasses` and connected logic and use only `main_class` attribute?

Reply:
I do believe so!

Reply:
Whenever they end up supporting more stuff that makes need to support `test_class`, we'll look into it. If that even happens :smile:

Reply:
cool